src/ring_theory/hahn_series.lean

```diff
@@ -126,6 +133,14 @@ support_single_subset h
 @[simp]
 lemma single_eq_zero : (single a (0 : R)) = 0 := (single a).map_zero
 
+lemma single_ne_zero (h : r ≠ 0) : single a r ≠ 0 :=
+begin
+  have h := support_single_of_ne h,
+  contrapose! h,
+  rw [h, support_zero, ne_comm],
+  exact (set.singleton_nonempty a).ne_empty,
+end
```

Comment:
Do we have some sort of `single_injective`? That would make this a 1-liner together with `single_eq_zero`.